jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/CreateAppRunnerServiceAction.kt

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.ecr.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.openapi.project.DumbAware
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import software.aws.toolkits.jetbrains.core.explorer.actions.SingleExplorerNodeAction
+import software.aws.toolkits.jetbrains.services.apprunner.ui.CreationDialog
+import software.aws.toolkits.jetbrains.services.ecr.EcrTagNode
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
+import software.aws.toolkits.resources.message
+
+class CreateAppRunnerServiceAction :
+    SingleExplorerNodeAction<EcrTagNode>(message("ecr.create.app_runner_service.action"), null, null),
+    DumbAware,
+    CoroutineScope by ApplicationThreadPoolScope("AppRunnerCreateServiceAction") {
+    override fun actionPerformed(selected: EcrTagNode, e: AnActionEvent) {
+        val project = e.getRequiredData(PlatformDataKeys.PROJECT)
+        launch {
+            withContext(getCoroutineUiContext()) {
+                CreationDialog(project, "${selected.repository.repositoryUri}:${selected.tag}").show()
+            }
+        }
```

Comment:
This is already on the UI thread, so the corotuine is not needed

Reply:
It's modeled off `software.aws.toolkits.jetbrains.services.apprunner.actions.CreateServiceAction`

Reply:
Huh...that one is wrong too :/

Reply:
moved both